forte/data/data_store.py

```diff
@@ -1365,9 +1408,10 @@ def within_range(
             if range_annotation is None:
                 yield from self.co_iterator_annotation_like(all_types)
             else:
-                for entry in self.co_iterator_annotation_like(all_types):
-                    if within_range(entry, range_annotation):
-                        yield entry
+                for entry_type in all_types:
+                    yield from get_bisect_range(
+                        entry_type, self.__elements[entry_type]
+                    )
```

Comment:
This actually changes the behavior since `co_iterator` is not used.

Reply:
In addition, consider adding/reusing test cases that work directly from the data pack where you can actually ingest some real data. The DataStore test cases actually use some sample data created manually, which might be inefficient.

Reply:
Actually I wrote this loop after looking at the working of `co_iterator` and tried to make it the same. Could you point me to where it might be different?

Reply:
I think `co-iterator` will yield the annotations based on the position, entry type is the secondary tie-breaker. But the new code will yield based on the types first, and then the position.

Reply:
In addition, we don't want the loop logic to be in two places (inside `co_iterator` and here). Instead, we should send `range_annotatoin` into `co_iterator` and handle it inside.

Reply:
Okay I get your point. I'll look into that. Also, for the test cases. I looked at a few existing test cases like [`predictor_test.py`](https://github.com/asyml/forte/blob/master/tests/forte/processors/predictor_test.py) where we ingest real data. The new methods works fine in that as well.

Reply:
1. Try to see if some data digestion steps are not in `predictor_test.py`, hopefully in `data_pack_test.py`. Since it will be easier to locate the correct test cases. `predictor_test` is not designed for this.
2. The fact it worked now (passing the tests) is a problem. Since this implementation is likely to change the behavior but not cached, we need a test case where you can see the difference between these two implementations